myriad-scheduler/src/main/java/com/ebay/myriad/Main.java

```diff
@@ -190,6 +196,29 @@ private void startNMInstances(Injector injector) {
       }
     }
 
+    /**
+     * Create NMProfile for any configured service
+     * @param injector
+     */
+    private void initJavaBasedTaskConfiguration(Injector injector) {
+      LOGGER.info("Initializing JavaBasedTaskConfiguration");
+      NMProfileManager profileManager = injector.getInstance(NMProfileManager.class);
+
+      Map<String, AuxTaskConfiguration> auxServicesConfigs = 
+          injector.getInstance(MyriadConfiguration.class).getAuxTaskConfigurations();
+      if (auxServicesConfigs != null) {
+        for (Map.Entry<String, AuxTaskConfiguration> entry : auxServicesConfigs.entrySet()) {
+          AuxTaskConfiguration config = entry.getValue();
+          Double cpu = (config.getCpus().isPresent() ? config.getCpus().get() : null);
+          Double mem = (config.getJvmMaxMemoryMB().isPresent() ? config.getJvmMaxMemoryMB().get() : null);
+          
```

Comment:
The idea of guava's optionals is to avoid null checks.  Feels like an anti pattern.  Also, guava has an `or` method which would let you replace the ternary statement with `Double cpus config.getCpus().or(SOME_VAL);`.